app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,102 @@
+class OrdersController < ApplicationController
+  before_action :find_user
+  before_action :find_uiq, only: [:index, :all]
+  before_action :find_order, except: [:index, :new, :create, :all]
+  before_action :findAddress, only: [:show]
+
+  def index
+    if @user.admin
+      @orders = Order.all
+    else
+      redirect_to user_orders_path(@user.id)
+    end
+  end
+
+  def show
+  end
+
+  def new
+    @order = Order.new
+  end
+  #
+  # def create
+  #   @order = Order.new(order_params)
+  #   if @order.save!
+  #     redirect_to # UNKNOWN
+  #   else
+  #     render # UNKNOWN
+  #   end
+  # end
+
+  def edit
```

Comment:
Yes I think the edit method here should be using session to find the user or using the filter to find the value for the @user attribute.